From 3bdf36dce303d5ba9ff46b5526cfc9cb9bd74c86 Mon Sep 17 00:00:00 2001
From: Emil M George <emilmgeorge@gmail.com>
Date: Sat, 24 May 2025 22:31:28 +0530
Subject: [PATCH 1/4] Use (pull) --rebase for updating plugins

---
 scripts/update_plugin.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/update_plugin.sh b/scripts/update_plugin.sh
index e533664..d99e189 100755
--- a/scripts/update_plugin.sh
+++ b/scripts/update_plugin.sh
@@ -21,7 +21,7 @@ pull_changes() {
 	local plugin="$1"
 	local plugin_path="$(plugin_path_helper "$plugin")"
 	cd "$plugin_path" &&
-		GIT_TERMINAL_PROMPT=0 git pull &&
+		GIT_TERMINAL_PROMPT=0 git pull --rebase &&
 		GIT_TERMINAL_PROMPT=0 git submodule update --init --recursive
 }
 
-- 
2.49.0

